### ISSUE TITLE
Unmount fusefs(sshfs/rclone) precizely on FreeBSD.

### ISSUE THREAD

**Issue by uvelichitel** [jarun/nnn#848]
**Is your feature request related to a problem? Please describe.**

FreeBDS does not have '_fusermount3_'. Contrary '_umount_(8)' used like `__APPLE__` does. So unmount with shortcut **u** doesn't work on FreeBSD.

**Describe the solution you'd like**
Should be fine to add rules in compilation conditions at src/nnn.c:4438 to be something like
 ```
static bool unmount(char *name, char *newpath, int *presel, char *currentpath)
{
#if defined (__APPLE__) || defined (__FreeBSD__)
	static char cmd[] = "umount";
#else
	static char cmd[] = "fusermount3"; /* Arch Linux utility */
	static bool found = FALSE;
#endif
	char *tmp = name;
	struct stat sb, psb;
	bool child = FALSE;
	bool parent = FALSE;
	bool hovered = TRUE;
	char mntpath[PATH_MAX];

#if !defined ( __APPLE__) && !defined (__FreeBSD__)
	/* On Ubuntu it's fusermount */
	if (!found && !getutil(cmd)) {
		cmd[10] = '\0';
		found = TRUE;
	}
#endif

	mkpath(cfgpath, toks[TOK_MNT], mntpath);

	if (tmp && strcmp(mntpath, currentpath) == 0) {
		mkpath(mntpath, tmp, newpath);
		child = lstat(newpath, &sb) != -1;
		parent = lstat(xdirname(newpath), &psb) != -1;
		if (!child && !parent) {
			*presel = MSGWAIT;
			return FALSE;
		}
	}

	if (!tmp || !child || !S_ISDIR(sb.st_mode) || (child && parent && sb.st_dev == psb.st_dev)) {
		tmp = xreadline(NULL, messages[MSG_HOSTNAME]);
		if (!tmp[0])
			return FALSE;
		hovered = FALSE;
	}

	/* Create the mount point */
	mkpath(mntpath, tmp, newpath);
	if (!xdiraccess(newpath)) {
		*presel = MSGWAIT;
		return FALSE;
	}

#if defined (__APPLE__) || defined (__FreeBSD__)
	if (spawn(cmd, newpath, NULL, F_NORMAL)) {
#else
	if (spawn(cmd, "-u", newpath, F_NORMAL)) {
#endif
		if (!xconfirm(get_input(messages[MSG_LAZY])))
			return FALSE;

#ifdef __APPLE__
		if (spawn(cmd, "-l", newpath, F_NORMAL)) {
#elif defined (__FreeBSD__)
		/* Force umount on FreeBSD */
		if (spawn(cmd, "-f", newpath, F_NORMAL)) { 
#else
		if (spawn(cmd, "-uz", newpath, F_NORMAL)) {
#endif
			printwait(messages[MSG_FAILED], presel);
			return FALSE;
		}
	}

	if (rmdir(newpath) == -1) {
		printwarn(presel);
		return FALSE;
	}

	return hovered;
}
```
I can do a patch or pull request.

**Comment by jarun** [comment]
Please raise the PR. Thanks!

**Comment by jarun** [comment]
Closed at #849.
